### ISSUE TITLE
1.30: wrong failures of assertions on mixed-cost balances

### ISSUE THREAD

**Issue by simonmichael** [simonmichael/hledger#2072]
This was hard to minimise, saving here for reference:

```
; i.j
; This balance assertion with a cost is broken by the balance assignment.
; Docs say costs in balance assertions are ignored.

2018-01-01
    (a)                                            1 A  = 1 A

2018-12-01
    (a)                                           -1 A @ $1 = 0 A
    (b)                                                     = 0  ; this breaks the above assertion
    ; adding an explicit amount fixes it:
    ; (b)                                          0        = 0
    ; removing a's cost fixes it:
    ; (a)                                         -1 A      = 0 A
    ; posting 0 A and asserting 1 A fixes it:
    ; (a)                                         -0 A @ $1 = 1 A

comment
$ watchexec -- 'echo; echo =========; hledger -f i.j reg a 2>&1 -I; echo; hledger -f i.j reg a 2>&1 | grep -Ev "^ " | head -8'

=========
2018-01-01                      (a)                            1 A           1 A
2018-12-01                      (a)                           -1 A             0

hledger: Error: /Users/simon/src/hledger/i.j:9:61:
8 | 2018-12-01

This balance assertion failed.
In account:    a
and commodity: A
this balance was asserted:     0
but the calculated balance is: 1
```

**Comment by simonmichael** [comment]
Probably related to #824.

Behaviour seems to have changed since 1.30.

**Comment by simonmichael** [comment]
Yet another, even earlier report of #2137 and #2150 (wrong failures of assertions on mixed-cost balances). Fixed in master.

**Comment by simonmichael** [comment]
This is fixed in 1.32.3.
